lib/livebook/runtime/evaluator.ex

```diff
@@ -475,17 +475,22 @@ defmodule Livebook.Runtime.Evaluator do
     end
 
     state = put_context(state, ref, new_context)
-
     output = Evaluator.Formatter.format_result(result, language)
 
+    definitions =
+      for {module, _} <- tracer_info.modules_defined,
+          definition = get_definition(context, module),
+          do: definition
+
     metadata = %{
       errored: error_result?(result),
       interrupted: interrupt_result?(result),
       evaluation_time_ms: evaluation_time_ms,
       memory_usage: memory(),
       code_markers: code_markers,
       identifiers_used: identifiers_used,
-      identifiers_defined: identifiers_defined
+      identifiers_defined: identifiers_defined,
+      definitions: definitions
```

Comment:
Let's update the typespecs in `Livebook.Runtime`, please add the missing `interrupted: boolean()` also :D

Reply:
Let's also add a test in `EvalutorTest`.